packages/preferences/src/browser/preferences-widget.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, multiInject } from "inversify";
+import { VirtualWidget } from "@theia/core/lib/browser/widgets/virtual-widget";
+import { Message } from "@phosphor/messaging";
+import { EditorPreferences } from "@theia/editor/lib/browser/editor-preferences";
+import { PreferenceService } from "../../../core/lib/browser/preferences";
+import { EditorManager } from "@theia/editor/lib/browser";
+import { ApplicationShell, PreferenceSchema, PreferenceScope } from "@theia/core/lib/browser";
+import { PreferenceProperty } from "@theia/core/lib/browser/preferences/preference-contribution";
+import { h } from '@phosphor/virtualdom';
+
+export interface PreferenceGroup {
+    name: string;
+    preferences: Preference[];
+    isExpanded: boolean;
+}
+
+export interface Preference {
+    name: string;
+    property: PreferenceProperty;
+}
+
+export class PreferencesWidget extends VirtualWidget {
```

Comment:
Would be good if this could use the tree widget. It would automatically look more aligned with the rest and we can make use of the fast filter, keyboard navigation and decoration support.